core/src/main/java/com/digitalpebble/storm/crawler/bolt/FetcherBolt.java

```diff
@@ -336,42 +324,61 @@ public synchronized FetchItemQueue getFetchItemQueue(String id) {
                 fiq = new FetchItemQueue(conf, customThreadVal, crawlDelay,
                         minCrawlDelay);
                 queues.put(id, fiq);
-
-                // Reset the cyclic iterator to start of the list.
-                it = Iterables.cycle(queues.keySet()).iterator();
             }
             return fiq;
         }
 
         public synchronized FetchItem getFetchItem() {
-
-            if (queues.isEmpty() || !it.hasNext())
+            if (queues.isEmpty()) {
                 return null;
+            }
 
             FetchItemQueue start = null;
 
             do {
-                FetchItemQueue fiq = queues.get(it.next());
+                Map.Entry<String, FetchItemQueue> nextEntry;
+                FetchItemQueue fiq;
+                Iterator i = queues.entrySet().iterator();
```

Comment:
> getFetchItem method is already synchronized.

of course

Would it make sense to create the iterator only once outside the do while loop? It would be more efficient and we would not need to keep track of the starting element at all?